plugins/modules/cloud/misc/terraform.py

```diff
@@ -449,12 +480,73 @@ def main():
     if state == 'present' and module.params.get('parallelism') is not None:
         command.append('-parallelism=%d' % module.params.get('parallelism'))
 
+    def format_args(vars):
+        if isinstance(vars, str):
+            return '"{string}"'.format(string=vars)
+        elif isinstance(vars, bool):
+            if vars:
+                return 'true'
+            else:
+                return 'false'
+        elif isinstance(vars, (integer_types, float)):
+            return str(vars)
+        else:
+            return vars
```

Comment:
Either always return `str(vars)`, like
```suggestion
        return str(vars)
```
or raise an error
```suggestion
        elif isinstance(vars, (integer_types, float)):
            return str(vars)
        raise TypeError('Unsupported type %s' % (type(vars), ))
```